### PR TITLE
Add future::try_select

### PR DESCRIPTION
**Pull request by taiki-e**
The same as [0.1's select2](https://docs.rs/futures/0.1.27/futures/future/trait.Future.html#method.select2).

### CODE REVIEW

**Comment by Matthias247** [comment]
Is there any situation you would prefer this over `select!`?
I would actually try to encourage people to really go the idiomatic async/await way, and not to offer multiple ways to do the same thing (that all need to be maintained).

**Comment by taiki-e** [comment]
`TrySelect` is `TryFuture` but `Select` is not.

File: futures-util/src/try_future/try_select.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+pub fn try_select<A, B>(future1: A, future2: B) -> TrySelect<A, B>
+    where A: TryFuture + Unpin, B: TryFuture + Unpin
+{
+    TrySelect { inner: Some((future1, future2)) }
+}
+
+impl<A: Unpin, B: Unpin> Future for TrySelect<A, B>
+    where A: TryFuture, B: TryFuture
+{
+    #[allow(clippy::type_complexity)]
```
Comment (cramertj) [inline futures-util/src/try_future/try_select.rs]: lol-- no kidding

**Comment by cramertj** [comment]
seems basically fine to me, and could help people move over code directly without going all the way to async/await. It is unfortunate to have all of these "multiple ways to do things" but after all that's sort of why `futures-util` exists.

**Pull request merged**